testing/internal/e2e/tests/base/scope_permissions_test.go

```diff
@@ -0,0 +1,174 @@
+	})
+
+	// Create new user
+	userId, err := boundary.CreateUserCli(t, ctx, "global")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("users", "delete", "-id", userId),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	err = boundary.SetAccountToUserCli(t, ctx, userId, accountId)
+	require.NoError(t, err)
+
+	// Find Ids of the roles to be modified
+	output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("roles", "list", "-format", "json"))
+	var rolesListResult roles.RoleListResult
+	err = json.Unmarshal(output.Stdout, &rolesListResult)
+	require.NoError(t, err)
+	var loginGrantsRoleId string
+	var authenticatedUserGrantsRoleId string
+	var loginAndDefaultGrantsRoleId string
+	for _, item := range rolesListResult.Items {
+		if item.Name == "Login Grants" {
+			loginGrantsRoleId = item.Id
+		}
+		if item.Name == "Authenticated User Grants" {
+			authenticatedUserGrantsRoleId = item.Id
+		}
+	}
+	output = e2e.RunCommand(ctx, "boundary", e2e.WithArgs("roles", "list", "-format", "json", "-scope-id", orgId))
+	err = json.Unmarshal(output.Stdout, &rolesListResult)
+	require.NoError(t, err)
+	for _, item := range rolesListResult.Items {
+		if item.Name == "Login and Default Grants" {
+			loginAndDefaultGrantsRoleId = item.Id
+		}
+	}
+
+	// In the org, prevent the user from being able to list all scopes
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs("roles", "remove-grants", "-id", loginAndDefaultGrantsRoleId, "-grant", "ids=*;type=scope;actions=list,no-op"))
+	require.NoError(t, output.Err, string(output.Stderr))
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs("roles", "add-grants", "-id", loginAndDefaultGrantsRoleId, "-grant", "ids=*;type=scope;actions=list"))
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// In global, remove the no-op permission from the login role
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs("roles", "remove-grants", "-id", loginGrantsRoleId, "-grant", "ids=*;type=scope;actions=list,no-op"))
+	require.NoError(t, output.Err, string(output.Stderr))
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs("roles", "add-grants", "-id", loginGrantsRoleId, "-grant", "ids=*;type=scope;actions=list"))
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("roles", "remove-grants", "-id", loginGrantsRoleId, "-grant", "ids=*;type=scope;actions=list"))
+		require.NoError(t, output.Err, string(output.Stderr))
+		output = e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("roles", "add-grants", "-id", loginGrantsRoleId, "-grant", "ids=*;type=scope;actions=list,no-op"),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	// In global, prevent the user from being able to read all specs
```

Comment:
I assume this is supposed to refer to scopes?
```suggestion
	// In global, prevent the user from being able to read all scopes
```

Reply:
Correct!